packages/veritone-json-schemas/src/index.test.js

```diff
@@ -0,0 +1,65 @@
+import { verifyObject, verifyTranscript, VALIDATORS } from './index';
+import _ from 'lodash';
+
+import glob from 'glob';
+
+const categories = {};
+
+// dynamically import all json files in examples or invalid-examples
+const files = glob.sync("schemas/vtn-standard/**/**(examples|invalid-examples)/!(master.json)//*.json");
+for(let file of files) {
+  const categoryIndex = file.split("/").findIndex(directory => _.includes(directory, "examples")) - 1;
+  const category = file.split("/")[categoryIndex];
+  let directory = file.split("/");
+  let fileName = directory.pop();
+  directory = directory.join("/");
+  const test = {directory, fileName};
+
+  if (category in categories) {
+    categories[category].push(test);
+  } else {
+    categories[category] = [test];
+  }
+}
+
+Object.keys(categories).forEach(category => {
+  describe(`"${category}" tests`, () => {
+    const validator = VALIDATORS[category];
+    if(typeof validator === 'function') {
+      for(let test of categories[category]) {
+        if( _.includes(test.directory, 'invalid' )) {
+
+          // JSON files in invalid-examples should not validate
+          it(`should NOT validate ${test.directory}/${test.fileName}`, async () => {
```

Comment:
I love that I can just add a new file in the proper folder/naming convention and it'll be picked up as a new test.  But the flip side of that of course is that the test errors are not very specific.  If there were any way to throw more specific error messages that'd be great.  If it's not easy, don't worry about it; the pros of having DRYer test case writing outweigh the cons of having to search for your errors IMO.

Reply:
The alternative to this would be to make the examples in a .JS vs .JSON file, and then provide something like

const data = {
   ...VTNSAMPLE
}

const errors = {
   ERRORS_CAPTURED_FROM_LIBRARY
}

but I'm preferring .json because it makes it easier for people to understand and easier for people to import. Also, I'm not sure how stable the error reporting is from version to version of AJV, so the test might be brittle if I do that.

It's always possible to add your own specific test in the test file if you want to test a specific scenario (which I will probably do for additional properties)

Reply:
Tradeoffs.  Sounds good.